foliage.cabal

```diff
@@ -38,32 +65,23 @@ executable foliage
         Foliage.Utils.GitHub
         Network.URI.Orphans
 
-    default-language:   GHC2021
+executable foliage
+    import:             shared-modules
+    main-is:            Main.hs
 
-    ghc-options:        -Wall
-    ghc-options:        -threaded
+library libfoliage
+    import:             shared-modules
```

Comment:
I don't think you want to do this. Rather, inline `shared-modules` into here, put the `Foliage` modules in `exposed-modules`, and add `foliage` to the `build-depends` of the executable.

Reply:
The idea of writing a library scares me a bit :P I'd be ok with just re-listing the `other-modules`. I am not sure if this is what @yvan-sraka was doing.

Reply:
What's wrong with a library? It's just a way of organizing the code really!

Reply:
Then I need to care about API changes :D There's no API if nobody else can use the code :D